### DISCUSSION TITLE
Suggestion: Include complexity level in `noExcessiveCognitiveComplexity` LSP diagnostic

### DISCUSSION THREAD

**Discussion by chrisgrieser** [biomejs/biome#4012] [Ideas]
- When running the biome cli, it reports: `Please refactor this function to reduce its complexity score from 17 to the max allowed complexity 15.`
- In the editor, biome reports: `Excessive complexity detected. (biome: lint/complexity/noExcessiveCognitiveComplexity)`

Basically, the LSP diagnostic is missing the allowed and current complexity level. However, when tackling such a complexity issue, it is often useful to know how much complexity has been reduced by a change you made etc. Thus, it would be very useful to include the complexity levels in the LSP diagnostic as well.

**Comment by Conaclos** [comment]
I agree. PR are welcome :)

**Comment by chrisgrieser** [comment]
haven't touched any rust code before, but this one change seems straightforward enough, so I made a PR: https://github.com/biomejs/biome/pull/4028

**Reply by chrisgrieser** [reply]
Pr has been merged
